test/extensions/filters/network/ext_authz/ext_authz_fuzz_test.cc

```diff
@@ -0,0 +1,123 @@
+#include "envoy/extensions/filters/network/ext_authz/v3/ext_authz.pb.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/network/address_impl.h"
+
+#include "extensions/filters/network/ext_authz/ext_authz.h"
+
+#include "test/extensions/filters/common/ext_authz/mocks.h"
+#include "test/extensions/filters/network/ext_authz/ext_authz_fuzz.pb.validate.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/mocks/network/mocks.h"
+#include "test/mocks/runtime/mocks.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::ReturnRef;
+using testing::WithArgs;
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ExtAuthz {
+
+Filters::Common::ExtAuthz::ResponsePtr
+makeAuthzResponse(Filters::Common::ExtAuthz::CheckStatus status) {
+  Filters::Common::ExtAuthz::ResponsePtr response =
+      std::make_unique<Filters::Common::ExtAuthz::Response>();
+  response->status = status;
+  return response;
+}
+
+Filters::Common::ExtAuthz::CheckStatus resultCaseToCheckStatus(
+    envoy::extensions::filters::network::ext_authz::Result::ResultSelectorCase result_case) {
+  Filters::Common::ExtAuthz::CheckStatus check_status;
+  switch (result_case) {
+  case envoy::extensions::filters::network::ext_authz::Result::kCheckStatusOk: {
+    check_status = Filters::Common::ExtAuthz::CheckStatus::OK;
+    break;
+  }
+  case envoy::extensions::filters::network::ext_authz::Result::kCheckStatusError: {
+    check_status = Filters::Common::ExtAuthz::CheckStatus::Error;
+    break;
+  }
+  case envoy::extensions::filters::network::ext_authz::Result::kCheckStatusDenied: {
+    check_status = Filters::Common::ExtAuthz::CheckStatus::Denied;
+    break;
+  }
+  default: {
+    // Unhandled status
+    PANIC("A check status handle is missing");
+  }
+  }
+  return check_status;
+}
+
+DEFINE_PROTO_FUZZER(const envoy::extensions::filters::network::ext_authz::ExtAuthzTestCase& input) {
+  try {
+    TestUtility::validate(input);
+  } catch (const ProtoValidationException& e) {
+    ENVOY_LOG_MISC(debug, "ProtoValidationException: {}", e.what());
+    return;
+  } catch (const ProtobufMessage::DeprecatedProtoFieldException& e) {
```

Comment:
quick question -- why not catch any EnvoyException that throws from `TestUtility::validate`?

Reply:
> quick question -- why not catch any EnvoyException that throws from `TestUtility::validate`?

Fixed. Thanks!